### PR TITLE
Create App: HMD missing functionalities 

### PR DESCRIPTION
**Pull request by AleziaKurdis**
This PR add a set of buttons **in HDM only** to allowed the following action:
**Copy - Cut - Paste - Duplicate**

This address Issue **#127 Add a “Duplicate” button on the Entity list of the “Create” application**

It also add 2 buttons for **Desktop** and **HMD**:
**Undo - Redo**

This PR address points 1 and 2 of the issue **#754 Create App: Misc. Improvement for HMD**

### CODE REVIEW

**Comment by AleziaKurdis** [comment]
Added a "Multiple Selection" activation button (HMD only)
to allow to do multiple selection (by the list) while we are working in HMD.
(This wasn't possible before)

**Comment by AleziaKurdis** [comment]
Added: 
-Parent and Unparent button on the toolbar of the list
-HMD multiselection applicable when we select directly inworld (in addition of from the list)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-89bdfd5.tar_ff39a03b.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-89bdfd5_ff39a03b.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-89bdfd5.tar_ff39a03b.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-89bdfd5_ff39a03b.exe

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-89bdfd5_ff39a03b.dmg

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-e15e1fc_a7ca92fb.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-e15e1fc.tar_a7ca92fb.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-e15e1fc.tar_a7ca92fb.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-e15e1fc_a7ca92fb.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-e15e1fc_a7ca92fb.exe

File: scripts/system/create/entitySelectionTool/entitySelectionTool.js [hunk 1]
```diff
@@ -315,6 +319,7 @@ SelectionManager = (function() {
             that.addChildrenEntities(originalEntityID, entitiesToDuplicate, entityHostTypes[i].entityHostType);
         }
         
+        var hasItFailed = false;
```
Comment (ctrlaltdavid) [inline scripts/system/create/entitySelectionTool/entitySelectionTool.js:322]: `hasFailed` might be a less awkward variable name?

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-a61087e.tar_37d694af.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-a61087e_37d694af.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-a61087e.tar_37d694af.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-a61087e_37d694af.exe

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-a61087e_37d694af.dmg

**Comment by SilverfishVR** [comment]
Wow, this is a big quality-of-life improvement for VR 💯 Thank you.

Everything seems to work as expected, once you understand the icons. 👍 
my only comment is that I feel like the order of buttons should be _cut copy paste_ rather than _copy cut paste_ it's just more "standard"

**Comment by AleziaKurdis** [comment]
This is now in that sequence: Cut Copy Paste.

I checked to add tooltips, but the standard ones don't work in HMD.
( I added this to my "to-do" list, if I found a way to do it I will do it in a future PR)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-454df7e_adf6652e.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-454df7e.tar_adf6652e.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-454df7e.tar_adf6652e.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-454df7e_adf6652e.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR760/Vircadia-Alpha-PR760-454df7e_adf6652e.exe

**Comment by SilverfishVR** [comment]
looks good 👍 
yes, I figured that tooltips would be out of scope for this PR since none of the other buttons have it.

**Comment by Aitolda** [comment]
Seems to work, although we greatly need some sort of tooltips when you hover your pointer over these. I had no idea what was what until I came back to this page.

**Comment by AleziaKurdis** [comment]
Tooltip are easy to add in desktop, but in HMD it's another story. And it's on the HMD button that it would be mainly useful.
This is on my to do list.

**Comment by Aitolda** [comment]
Yeah. I realize it's probably an undertaking.  Just thinking about the grand scheme.

**Pull request merged**